datastore/datastore_manager.go

```diff
@@ -26,11 +26,23 @@ import (
 
 // Options holds settings specific to datastore
 type Options struct {
+	// Project sets the GCP project to use for communicating to datastore.
+	Project string
 }
 
 // New initializes the Manager struct implementing a generic storage.Manager interface
 func New(ctx context.Context, options *Options) *Manager {
-	dsClient, err := datastore.NewClient(ctx, fetchProjectID())
+	if options.Project == "" {
+		if env.IsAppEngine() {
+			options.Project = env.AppEngineProject()
+			log.Infof("No datastore project specified, defaulting to GAE project: %v", options.Project)
+		} else {
+			log.Fatalf(
+				"Could not determine project to use for Datastore, please set DATASTORE_PROJECT or --datastore-project flag")
+		}
+	}
```

Comment:
Since Go style recommends minimizing vertical space and indent levels, switch the logic around so that there's no `else` block:

```go
if options.Project == "" {
  if !env.IsAppEngine() {
    log.Fatalf("Could not determine project to use for Datastore, please set DATASTORE_PROJECT or --datastore-project flag")
  }
  options.Project = env.AppEngineProject()
  log.Infof("No datastore project specified, defaulting to GAE project: %v", options.Project)
}
```

Also, no need to line wrap the log.Fatalf line.

Reply:
Agreed on line wrap.

I would prefer leaving the env.IsAppengine however to follow the "positive boolean" pattern: https://engdoc.corp.google.com/eng/doc/tott/episodes/530.md?cl=head

WDYT?

Reply:
I see the reason. Just know if this ever goes to Go Readability approval team, they'll insist on it being the other way around.

Reply:
Ack. This is good to know. Thanks!